### PR TITLE
Send unknown event when there is no auth event.

### PR DESCRIPTION
**Pull request by yuchenshi**
### Description

This brings Auth Emulator iframe a bit closer to production, where it always sends something on start (which may be an `unknown` event if there is nothing stored).

If using iframe messaging, it may send `unknown` first then immediately followed by an `authEvent`. I have no idea whether this works with the JS/TS SDK.

### Scenarios Tested

NOT TESTED YET

### Sample Commands

N/A

### CODE REVIEW

File: src/emulator/auth/handlers.ts [hunk 1]
```diff
@@ -249,28 +249,26 @@ export function registerHandlers(
...
     parentContainer.send('authEvent', {
-      type: 'authEvent',
-      authEvent: authEvent,
+      type: authEvent ? 'authEvent' : 'unknown',
+      authEvent: authEvent || {},
     }, function(responses) {
       if (!responses || !responses.length ||
-          !responses[responses.length - 1].status === 'ACK') {
+          responses[responses.length - 1].status !== 'ACK') {
         return alert("Auth Emulator Internal Error: Sending authEvent failed.");
```
Comment (samhorlbeck) [inline src/emulator/auth/handlers.ts:272]: This throws an alert up whenever the iframe is initialized

Comment (yuchenshi) [reply src/emulator/auth/handlers.ts]: Reverted for now. This condition is really messed up and misleading, and I'll look into it later.

Comment (samhorlbeck) [reply src/emulator/auth/handlers.ts]: Okay cool let me play with it tomorrow but otherwise lgtm :+1:

Comment (samhorlbeck) [reply src/emulator/auth/handlers.ts]: This works in my testing, thanks again!

**Review by samhorlbeck — APPROVED** [review]

File: src/emulator/auth/handlers.ts [hunk 2]
```diff
@@ -249,25 +249,23 @@ export function registerHandlers(
...
+    delete sessionStorage['firebase:redirectEvent:' + storageKey];
   }
 
-  function sendAuthEvent(parentContainer, authEvent) {
+  function sendAuthEvent(authEvent) {
     parentContainer.send('authEvent', {
-      type: 'authEvent',
-      authEvent: authEvent,
+      type: authEvent ? 'authEvent' : 'unknown',
+      authEvent: authEvent || {},
```
Comment (yuchenshi) [inline src/emulator/auth/handlers.ts:268]:
```suggestion
      authEvent: authEvent,
      error: {code: 'auth/no-auth-event'},
```

File: src/emulator/auth/handlers.ts [hunk 3]
```diff
@@ -249,25 +249,24 @@ export function registerHandlers(
...
+    delete sessionStorage['firebase:redirectEvent:' + storageKey];
   }
 
-  function sendAuthEvent(parentContainer, authEvent) {
+  function sendAuthEvent(authEvent) {
     parentContainer.send('authEvent', {
-      type: 'authEvent',
+      type: authEvent ? 'authEvent' : 'unknown',
       authEvent: authEvent,
+      error: {code: 'auth/no-auth-event'},
```
Comment (yuchenshi) [inline src/emulator/auth/handlers.ts:269]:
```suggestion
      error: authEvent ? undefined : {code: 'auth/no-auth-event'},
```

File: src/emulator/auth/handlers.ts [hunk 4]
```diff
@@ -249,25 +249,24 @@ export function registerHandlers(
...
+    sendAuthEvent(authEvent);
+    delete sessionStorage['firebase:redirectEvent:' + storageKey];
   }
 
-  function sendAuthEvent(parentContainer, authEvent) {
+  function sendAuthEvent(authEvent) {
     parentContainer.send('authEvent', {
-      type: 'authEvent',
+      type: authEvent ? 'authEvent' : 'unknown',
       authEvent: authEvent,
```
Comment (yuchenshi) [inline src/emulator/auth/handlers.ts:268]:
```suggestion
      authEvent: authEvent || {},
```

File: src/emulator/auth/handlers.ts (hunk 1 above)
Comment (yuchenshi) [reply src/emulator/auth/handlers.ts]: Fix for the condition: https://github.com/firebase/firebase-tools/pull/3188

**Pull request merged**